### ISSUE TITLE
.NET SDK not found

### ISSUE THREAD

**Issue by wlewis55** [microsoft/vscode-dotnettools#532]
### Describe the Issue

Type: Bug

I have been using the official Unity support since it was released, and I have run into an error multiple times that I cannot seem to fix. With all of the correct versions (Unity 2022.3.2f1, Unity Visual Studio Editor 2.0.20, no Unity VS Code Editor package, VSCode Unity Support 0.9.0, C# Dev Kit 0.3.21, C# 2.0.357) I continue to see a pop up that says:

Failed at 'command: "dotnet.exe" restore <file_path>.sln --interactive' (382ms) with error: ERROR ExitCode(2147516561)
Details
The command could not be loaded, possibly because:
  * You intended to execute a .NET application:
      The application 'restore' does not exist.
  * You intended to execute a .NET SDK command:
      No .NET SDKs were found.
This causes intellisence to fail, making VSCode useless for Unity development. I have verified that my .NET SDK's are installed properly.

### **CURRENT FIX:**
[How to fix "No .NET SDKs were found." error--VSCode](https://stackoverflow.com/questions/73247933/how-to-fix-no-net-sdks-were-found-error-vscode)

Deleting everything in the C:\Program Files (x86)\dotnet folder and reloading the VSCode window allows for a Completed command: "dotnet.exe" restore <file_path>.sln --interactive (15270ms)

_Note: This may be a duplicate of #527_ 

### Environment Information

VS Code version: Code 1.81.1 (https://github.com/microsoft/vscode/commit/6c3e3dba23e8fadc360aed75ce363ba185c49794, 2023-08-09T22:22:42.175Z)
OS version: Windows_NT x64 10.0.19044

**Comment by jbevain** [comment]
@wlewis55 hi! Thanks for reaching out!

Is there any chance you could try with the latest previews of the `C#` and `C# Dev Kit` extensions as they've shipped a few fixes recently related to finding and loading the SDK.

I'm currently running:

* `C#`: 2.2.10
* `C# Dev Kit`: 0.5.47

**Comment by kvenkatrajan** [comment]
@wlewis55, Hi, What version of the .NET SDK do you have installed?

**Comment by yehorbk** [comment]
Hi! I have the same problem. I'm working on a M1 Mac where I have the latest stable MacOS (13.2) and .NET (7.0.401) versions installed. I tried both the stable and the pre-release C# and C# Dev Kit extensions versions and nothing did help.

In the `OUTPUT` section it says that the `Language server initialized`, but then I get the next error from the C# Dev Kit:

```
No installed .NET SDK was found on the computer. If it
is installed, make sure it can be found through the
PATH environment variable. C# Dev Kit does not work
without a supported version of SDK. After installing
the SDK, you need restart the VS Code session, or the
computer.
```

Also I have the same message in the `Solution Explorer` window when I try to open a project.

**Comment by tmeschter** [comment]
Moving to "area-project-cps" as this concerns finding the SDK.

**Comment by Michael-Eng** [comment]
@yehorbk and @wlewis55 is dotnet.exe in your path? Can you check from within the terminal inside VS Code?

We know there are some instances where installing .NET must happen before VS Code is launched so dotnet.exe is added to the path before the process starts.

**Comment by yehorbk** [comment]
@Michael-Eng
Yes, it's there, I can run `dotnet --info` as example:

<img width="477" alt="dotnet-info" src="https://github.com/microsoft/vscode-dotnettools/assets/35728917/777f7139-e012-48d1-ad2f-b2719d291520">

And in fact, I don't get this error anymore. I restarted computer several times since the last test and have installed the Visual Studio IDE, so that may have been what did the trick.

By the way, after this I got another error:

```bash
Created Test Controller
Using vstest from dotnet sdk in [/usr/local/share/dotnet/sdk/7.0.401].
unable to create test controller for c# extension: Error: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/7.0.401/vstest.console.dll" /parentprocessid:2717 /port:49394.
System.Exception: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/7.0.401/vstest.console.dll" /parentprocessid:2717 /port:49394.
 ---> System.ComponentModel.Win32Exception (2): An error occurred trying to start process '/usr/local/bin/dotnet' with working directory '/path/to/project'. No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
   --- End of inner exception stack trace ---
   at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProviders.LocalTestPlatformProvider.ActivateSessionAsync(ITestWindowCallbackService testWindowCallbackService, ITestAgentBridgeCallback TestAgentBridgeCallback, Boolean enableDiagnosticLogging, RuntimeEnvironment runtimeEnvironment, IDictionary`2 environment, IDictionary`2 props, InitializeCapabilitiesRequest clientCapabilities, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformAgentPool.<>c__DisplayClass24_0.<<CreateRemoteAgentAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.TestWindow.Utilities.EventPumpExtensions.<>c__DisplayClass4_0`1.<<EnqueueAsync>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.TestWindow.Extensibility.TaskCompletionSourceExtensions.RunTaskWithCatchAsync[T](TaskCompletionSource`1 taskCompletionSource, Func`1 task)
   at Microsoft.VisualStudio.TestWindow.Logging.ILoggerExtensions.CallWithCatchAsync(IInternalLogger log, Func`1 func)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProvider.GetProviderAsync(ITestWindowCallbackService callbackClient, IInternalLogger logger, ITestWindowHostTelemetry telemetryService, ICollection`1 runtimeEnvironments, IDictionary`2 environmentVariables, Boolean enableDiagnosticLogging, Boolean autoInstallRuntimePrereqs, InitializeCapabilitiesRequest clientCapabilities, ITestPlatformProviderMap testPlatformProviderMap)
   at Microsoft.VisualStudio.TestWindow.Host.TestWindowStoreService.InitializeVSCodeServiceAsync(InitializeOptions initializeRequest, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.TestWindow.VSCode.Service.TestWindowVSCodeService.InitializeAsync(InitializeOptions options, CancellationToken cancellationToken)
```

But this seems to be a common issue for MacOS. The fix with overwriting the link from issue #384 did help me, so now I have everything up and running 🏄‍♂️.

P.S. I'll try to uninstall Visual Studio IDE to see if this breaks VS Code Extensions.

**Comment by Michael-Eng** [comment]
Thanks @yehorbk. The vsest issue is different and probably covered in #384.

**Comment by imokoi** [comment]
`visualstudiotoolsforunity.vstuc requested to download the .NET Runtime.
Downloading .NET version(s) 7.0.11~arm64 ...................................................................................................................................................................................................................`

It appeared to me, Mac. I have not installed visual studio IDE.

**Comment by Michael-Eng** [comment]
@imokoi this appears to be a VS for Unity component

**Comment by jbevain** [comment]
We're currently investigating the issue where sometimes, on macOS, you'll get a prompt telling you that no .NET SDK has been found, which prevents the `C#` and `C# Dev Kit` extension from working properly.

As a workaround, you can add `dotnet` to your PATH directly:

`launchctl setenv PATH $PATH:/usr/local/share/dotnet`

Then restart Unity Hub/Unity/Visual Studio Code.

**Comment by jbevain** [comment]
Tracking VS Code issue there: https://github.com/microsoft/vscode/issues/194736

@wlewis55 did the workaround above help? Thanks!

**Comment by retteghy** [comment]
this issue is not solved for me, despite having installed both dotnet 6 and dotnet 7 on Mac M1 and executing
`launchctl setenv PATH $PATH:/usr/local/share/dotnet `
VS Code still tries downloading it again (but that keeps failing, as I am in China). 
`Downloading .NET version(s) 7.0.13~arm64 ...... `
I have following dotnet versions installed 
```
dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.24 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.13 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.19 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.24 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
```

**Comment by Michael-Eng** [comment]
@retteghy what does 'dotnet.exe --list-sdks' show? We need components in the SDK since you're building an app.

How did you install .NET? We need to be able to find dotnet.exe and most problems have been when it's installed somewhere we cannot find it. 

Is dotnet.exe on your $PATH? 

Other users have told us that running 'brew remove dotnet' and then restarting VS Code (closing the app, not closing the window) has fixed it for them.

**Comment by lifengl** [comment]
Hey, @yehorbk : it is likely when you installed VS, it installed .NET SDK to the machine, and when you uninstalled it, the SDK might be removed? The C#/C# Dev Kit VS code extension does not have any function to install SDK. It does automatically download .NET runtime, which might lead some confusion here.  The .NET runtime downloaded by the extension is just to support the extension to run (only to boot the extension itself), it does not include any SDK required to build/open any .NET project. SDK must be manually installed (and its version must match what the project requires) before any project can be opened/worked correctly.  If you cannot build/restore projects in a pure command window/shell, it is likely you would not get them to work in the vscode extension.  This is different from VS, which is a more complete tool set.

Also for the extension to find SDK installed, it must be in the PATH environment.  In some cases, a developer might install multiple SDKs on the machine, only the first one resolved in the PATH environment would be used.  In some environments, SDKs for different architectures could be installed on the same machine, for example both x86/x64 installed on windows machine, or ARM/Intel based SDKs installed on Mac M1/M2 machines.  The extension which is built to native architecture of the machine can only work with the SDK matching it (because they would need be loaded into same process, so no simulation layer could help). If both SDKs are in PATH environment, they must be in the correct order to work.  It would explain the workaround mentioned by @wlewis55 in the beginning of this thread. And for the earlier issue, it is likely no valid x64 SDK is installed on the machine, which has to be fixed manually. (As the extension does not install them automatically.)

For @retteghy: dotnet runtime is not enough for you to develop applications, it is to support pre-built applications depending on them to run in your environments, so they are not for developers. you have to install full SDKs to develop applications, and the proper way to check that is to use dotnet --list-sdks. As it is mentioned earlier, you need install SDKs manually before using those VS code extensions.  If you just install them, you might need reboot the machine to ensure environment variables are updated. Also, if you installed NET 7 SDK to your machine, the C# Dev Kit would use it, and would not start to download the runtime required to boot the extension.

**Comment by ChaimHong** [comment]
use unity + vscode （c# dev kit, c#, unity)
one unity project is ok, anther unity project is not work , say "sdk not found"
```
# dotnet --info
.NET SDK:
 Version:           8.0.100
 Commit:            57efcf1350
 Workload version:  8.0.100-manifests.6c33ef20

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  13.5
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/8.0.100/

.NET workloads installed:
 Workload version: 8.0.100-manifests.6c33ef20
There are no installed workloads to display.

Host:
  Version:      8.0.0
  Architecture: arm64
  Commit:       5535e31a71

.NET SDKs installed:
  7.0.404 [/usr/local/share/dotnet/sdk]
  8.0.100 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.14 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.14 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  x64   [/usr/local/share/dotnet/x64]
    registered at [/etc/dotnet/install_location_x64]

Environment variables:
  DOTNET_ROOT       [/usr/local/share/dotnet]

global.json file:
  Not found
  ```
  dev kit output
  ```
  Starting Spawn .NET server...
Starting Open a solution...
Starting Open a solution with environment service...
Starting Clear environment...
Using local .NET runtime at "/Users/xxxxx/Library/Application Support/Code/User/globalStorage/ms-dotnettools.vscode-dotnet-runtime/.dotnet/7.0.14~arm64/dotnet"
.NET server started and IPC established in 1080ms
Completed Spawn .NET server (1099ms)
Completed Clear environment (1275ms)
Completed Open a solution with environment service (1346ms)
Starting Restore solution...
Completed Open a solution (1352ms)
Starting NuGet restore for the solution.
Starting command: "dotnet" restore /Users/xxxxx/github.com/chexiongsheng/puerts_unity_demo/projects/1_Start_Template/1_Start_Template.sln --interactive...
[object Object]
[object Object]
```

**Comment by jbevain** [comment]
> use unity + vscode （c# dev kit, c#, unity) one unity project is ok, anther unity project is not work , say "sdk not found"

That sounds like https://github.com/microsoft/vscode/issues/194736

**Comment by ericdrobinson** [comment]
I [mentioned this](https://github.com/microsoft/vscode/issues/194736#issuecomment-1849129491) over in #194736, but if you are using Unity on macOS, openning your C# project from Unity's menu ("Open C# Project") seems to cause this issue.

Instead, create a new VS Code window from within VS Code and then open the project using "Open Folder" from within VS Code directly. That "resolved" the issue for me.

_[I put "resolved" in quotes because it's actually just a workaround..._

**Comment by webreidi** [comment]
FYI, we had another Mac user, doing basic C# learning tasks (not Unity) hit this. He was able to do the first 3 tasks with no issues. He closed down for the night and came back to do the 4th tasks and his SDK was no longer available to C# Dev Kit. He ran the "launchctl setenv PATH $PATH:/usr/local/share/dotnet" command and it got everything working again. 

Any ideas?

**Comment by Michael-Eng** [comment]
@webreidi Based on your description, I think this is a .NET install issue where .NET isn't available on the path. Maybe follow up with @nagilson but we need to be able to find .NET.

Jb did find one issue in VS Code based on the way it was launched but this specific issue seems to be that the path wasn't set up permanently?

**Comment by nagilson** [comment]
I agree with @Michael-Eng, and once the global API is used, that installer will set the PATH, so I dont see this issue happening once that is addded.

**Comment by burkeholland** [comment]
I'm getting this on a Mac M1 with a standard Azure Functions .NET 6 project - nothing fancy...

Version: 1.87.0
Commit: 019f4d1419fbc8219a181fab7892ebccf7ee29a2
Date: 2024-02-27T23:42:51.279Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.3.0

C# DevKit: v1.3.10

My dotnet location is a bit different because homebrew so `/opt/homebrew/bin/dotnet`. Running `launchctl setenv PATH $PATH:/opt/local/share/dotnet` has no effect.

Output from C# DevKit...

````
Starting Spawn .NET server...
Starting Open a solution...
Starting Open a solution with environment service...
Starting Clear environment...
Using local .NET runtime at "/Users/burkeholland/Library/Application Support/Code/User/globalStorage/ms-dotnettools.vscode-dotnet-runtime/.dotnet/7.0.16~arm64/dotnet"
.NET server started and IPC established in 17140ms
Completed Spawn .NET server (17729ms)
Completed Clear environment (18177ms)
Completed Open a solution with environment service (18282ms)
Starting Restore solution...
Completed Open a solution (18289ms)
Starting NuGet restore for the solution.
Starting command: "dotnet" restore /Users/burkeholland/dev/the-urlist/migration-tool/migration-tool.sln --interactive...
[object Object]
[object Object]
````

**Comment by I2olanD** [comment]
Same for me, uninstalled it with brew and installed via the .NET installer:

```
.NET SDK:
 Version:           8.0.201
 Commit:            4c2d78f037
 Workload version:  8.0.200-manifests.3097af8b

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  14.3
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/8.0.201/

.NET workloads installed:
There are no installed workloads to display.

Host:
  Version:      8.0.2
  Architecture: arm64
  Commit:       1381d5ebd2

.NET SDKs installed:
  8.0.201 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  x64   [/usr/local/share/dotnet/x64]
    registered at [/etc/dotnet/install_location_x64]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Comment by Timo972** [comment]
I am facing the same problem on Mac with the SDK installed using the ``dotnet-sdk-8.0.203-osx-arm64.pkg`` installer.

```
Failed to find dotnet info from path, falling back to acquire runtime via ms-dotnettools.vscode-dotnet-runtime
Error running dotnet --info: Error: Command failed: dotnet --info
/bin/sh: dotnet: command not found


/bin/sh: dotnet: command not found
```

``/usr/local/share/dotnet`` is present in PATH

```
.NET SDK:
 Version:           8.0.203
 Commit:            5e1ceea679
 Workload version:  8.0.200-manifests.4e94be9c

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  14.4
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/8.0.203/

.NET workloads installed:
There are no installed workloads to display.

Host:
  Version:      8.0.3
  Architecture: arm64
  Commit:       9f4b1f5d66

.NET SDKs installed:
  6.0.413 [/usr/local/share/dotnet/sdk]
  6.0.417 [/usr/local/share/dotnet/sdk]
  6.0.420 [/usr/local/share/dotnet/sdk]
  7.0.307 [/usr/local/share/dotnet/sdk]
  7.0.311 [/usr/local/share/dotnet/sdk]
  7.0.314 [/usr/local/share/dotnet/sdk]
  8.0.100 [/usr/local/share/dotnet/sdk]
  8.0.203 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.21 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.25 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.28 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.14 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.17 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.3 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.25 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.28 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.14 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.17 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.3 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Comment by nagilson** [comment]
> @nagilson do we have a solution for Mac folks who use homebrew to install the SDK? Is there a way to get it working?

I'm not sure we do, unfortunately. @baronfel Do you know if this is a supported scenario, I don't think it is? My advice would be to use the pkgs because those will set the path, for homebrew the user might need to manually set it.

**Comment by nagilson** [comment]
> I am facing the same problem on Mac with the SDK installed using the `dotnet-sdk-8.0.203-osx-arm64.pkg` installer.
> 
> ```
> Failed to find dotnet info from path, falling back to acquire runtime via ms-dotnettools.vscode-dotnet-runtime
> Error running dotnet --info: Error: Command failed: dotnet --info
> /bin/sh: dotnet: command not found
> 
> 
> /bin/sh: dotnet: command not found
> ```
> 
> `/usr/local/share/dotnet` is present in PATH
> 
> ```
> .NET SDK:
>  Version:           8.0.203
>  Commit:            5e1ceea679
>  Workload version:  8.0.200-manifests.4e94be9c
> 
> Runtime Environment:
>  OS Name:     Mac OS X
>  OS Version:  14.4
>  OS Platform: Darwin
>  RID:         osx-arm64
>  Base Path:   /usr/local/share/dotnet/sdk/8.0.203/
> 
> .NET workloads installed:
> There are no installed workloads to display.
> 
> Host:
>   Version:      8.0.3
>   Architecture: arm64
>   Commit:       9f4b1f5d66
> 
> .NET SDKs installed:
>   6.0.413 [/usr/local/share/dotnet/sdk]
>   6.0.417 [/usr/local/share/dotnet/sdk]
>   6.0.420 [/usr/local/share/dotnet/sdk]
>   7.0.307 [/usr/local/share/dotnet/sdk]
>   7.0.311 [/usr/local/share/dotnet/sdk]
>   7.0.314 [/usr/local/share/dotnet/sdk]
>   8.0.100 [/usr/local/share/dotnet/sdk]
>   8.0.203 [/usr/local/share/dotnet/sdk]
> 
> .NET runtimes installed:
>   Microsoft.AspNetCore.App 6.0.21 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 6.0.25 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 6.0.28 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 7.0.14 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 7.0.17 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 8.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 8.0.3 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.NETCore.App 6.0.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 6.0.25 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 6.0.28 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 7.0.14 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 7.0.17 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 8.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 8.0.3 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
> 
> Other architectures found:
>   None
> 
> Environment variables:
>   Not set
> 
> global.json file:
>   Not found
> 
> Learn more:
>   https://aka.ms/dotnet/info
> 
> Download .NET:
>   https://aka.ms/dotnet/download
> ```

The first failure you are seeing is related to failing to find a runtime and not an SDK. May you see if there is a output window for '.NET Runtime' and what that says. if anything?

**Comment by nagilson** [comment]
> > The first failure you are seeing is related to failing to find a runtime and not an SDK. May you see if there is a output window for '.NET Runtime' and what that says. if anything?
> 
> There's nothing at all being printed to the .Net Runtime Output.
> 
> **EDIT** I had this issue for over a month and suddenly it resolved by itself after posting here. 😄

Ok cool, well im glad its fixed 😋

**Comment by nagilson** [comment]
BTW, the .NET SDK can be installed by C# DevKit now in the walkthrough section in the prerelease versions. For people facing issues on Mac, I would suggest they try that method of SDK installation. :)

**Comment by pszponder** [comment]
I have installed dotnet using [mise](https://mise.jdx.dev/), an alternative to tools such as [asdf](https://asdf-vm.com/) to manage multiple different program runtimes.

I too am getting an error that the sdk is not being found. 
![Screenshot from 2024-04-20 09-04-01](https://github.com/microsoft/vscode-dotnettools/assets/14116413/bad31f3f-fcb6-40ab-943e-ec7b091754d7)

This is strange to me because I do have dotnet installed and I can invoke it through the command line just fine.

![image](https://github.com/microsoft/vscode-dotnettools/assets/14116413/0ea2afa2-e8e4-4334-8385-d41b23382305)

![image](https://github.com/microsoft/vscode-dotnettools/assets/14116413/0d3a3708-081d-4482-a7cf-ed383d8a3b8d)

I've tried setting paths for dotnet within VSCode but no luck there (maybe I'm using the wrong path?)

![image](https://github.com/microsoft/vscode-dotnettools/assets/14116413/4632c2c1-44b2-4ce8-b8da-3a5186f8f41a)

**Comment by nagilson** [comment]
If you set the existing dotnet path for the C# extension ID (different setting shown in your screenshot) does it work?

**Comment by pszponder** [comment]
Do you mean something like this?

```
"dotnetAcquisitionExtension.existingDotnetPath": [
  {
    "extensionId": "msazuretools.azurerm-vscode-tools",
    "path": "/home/piotr/.local/share/mise/installs/dotnet/8.0.204/sdk/8.0.204/"
  }
],
```

This what I tried but it does not seem to help.

**Comment by pszponder** [comment]
Actually, I think I fixed it by adding this to my .zprofile:

`export DOTNET_ROOT="$(dirname $(which dotnet))"`

**Comment by prangel-stuller** [comment]
> Actually, I think I fixed it by adding this to my .zprofile:
> 
> `export DOTNET_ROOT="$(dirname $(which dotnet))"`

I'll try to remember to come back to this in a few weeks if I see otherwise, but on first pass this appears to be a reliable workaround for the issue.

**Comment by toihocly** [comment]
@all, Hello All, for people have same issue with it. 
Study case:
after install 
dotnet-sdk-8.0.402-osx-x64.pkg and C# Dev Kit in VScode it doesn't work. the issue display with ".NET SDK not found..."

workaround based on all comments above:

step 1: run this code
> launchctl setenv PATH $PATH:/usr/local/share/dotnet

step 2: open ~/.zshrc and add this code
> export PATH=$PATH:/usr/local/share/dotnet

open vscode, reload window and check it again. if it doesn't work. follow next step. (close all vscode window)

step 3: google download new vscode on this https://code.visualstudio.com/

step 4: open new vscode. the problem will be solved

This is my workaround. if you have time you can check and report more for every know.

**Comment by fuunnx** [comment]
Hi, same issue on windows, after a while I tried those steps :

> step 3: google download new vscode on this https://code.visualstudio.com/
> step 4: open new vscode. the problem will be solved

and indeed it worked, thank you !

**Comment by Christiaan-Mathu** [comment]
> step 3: google download new vscode on this https://code.visualstudio.com/
> 
> step 4: open new vscode. the problem will be solved

At first I did not think this would solve my issue but it did thank you 😄

**Comment by nouhalashqar** [comment]
> [@wlewis55](https://github.com/wlewis55) hi! Thanks for reaching out!
> 
> Is there any chance you could try with the latest previews of the `C#` and `C# Dev Kit` extensions as they've shipped a few fixes recently related to finding and loading the SDK.
> 
> I'm currently running:
> 
> * `C#`: 2.2.10
> * `C# Dev Kit`: 0.5.47
